prow/github/types.go

```diff
@@ -1090,20 +1093,81 @@ type Milestone struct {
+	Parents     []GitCommit `json:"parents,omitempty"`
+	HTMLURL     string      `json:"html_url,omitempty"`
+	URL         string      `json:"url,omitempty"`
+	CommentsURL string      `json:"comments_url,omitempty"`
+
+	// Details about how many changes were made in this commit. Only filled in during GetCommit!
+	Stats *CommitStats `json:"stats,omitempty"`
+	// Details about which files, and how this commit touched. Only filled in during GetCommit!
+	Files []CommitFile `json:"files,omitempty"`
+}
+
+// CommitStats represents the number of additions / deletions from a file in a given RepositoryCommit or GistCommit.
+type CommitStats struct {
+	Additions int `json:"additions,omitempty"`
+	Deletions int `json:"deletions,omitempty"`
+	Total     int `json:"total,omitempty"`
+}
+
+// CommitFile represents a file modified in a commit.
+type CommitFile struct {
+	SHA              string `json:"sha,omitempty"`
+	Filename         string `json:"filename,omitempty"`
+	Additions        int    `json:"additions,omitempty"`
+	Deletions        int    `json:"deletions,omitempty"`
+	Changes          int    `json:"changes,omitempty"`
+	Status           string `json:"status,omitempty"`
+	Patch            string `json:"patch,omitempty"`
+	BlobURL          string `json:"blob_url,omitempty"`
+	RawURL           string `json:"raw_url,omitempty"`
+	ContentsURL      string `json:"contents_url,omitempty"`
+	PreviousFilename string `json:"previous_filename,omitempty"`
 }
 
 // GitCommit represents a GitHub commit.
 type GitCommit struct {
-	SHA     string `json:"sha,omitempty"`
-	Message string `json:"message,omitempty"`
+	SHA          string                 `json:"sha,omitempty"`
+	Author       CommitAuthor           `json:"author,omitempty"`
+	Committer    CommitAuthor           `json:"committer,omitempty"`
+	Message      string                 `json:"message,omitempty"`
+	Tree         Tree                   `json:"tree,omitempty"`
+	Parents      []GitCommit            `json:"parents,omitempty"`
+	Stats        *CommitStats           `json:"stats,omitempty"`
+	HTMLURL      string                 `json:"html_url,omitempty"`
+	URL          string                 `json:"url,omitempty"`
+	Verification *SignatureVerification `json:"verification,omitempty"`
+	NodeID       string                 `json:"node_id,omitempty"`
+
+	// CommentCount is the number of GitHub comments on the commit. This
+	// is only populated for requests that fetch GitHub data like
+	// Pulls.ListCommits, Repositories.ListCommits, etc.
+	CommentCount *int `json:"comment_count,omitempty"`
+}
+
+// CommitAuthor represents the author or committer of a commit. The commit
+// author may not correspond to a GitHub User.
+type CommitAuthor struct {
+	Date  time.Time `json:"date,omitempty"`
+	Name  string    `json:"name,omitempty"`
+	Email string    `json:"email,omitempty"`
+
+	// The following fields are only populated by Webhook events.
+	Login string `json:"username,omitempty"`
```

Comment:
Please make this a pointer as its optional

Reply:
Sorry for that, I'm  too careless.